ci/unit/test_main_node.sh

```diff
@@ -24,7 +24,17 @@ sudo ln -s "$SL_PREFIX/include"  /usr/share/spdk/include
 
 # set CMOCKA envs here
 export CMOCKA_MESSAGE_OUTPUT=xml
-export CMOCKA_XML_FILE="$DAOS_BASE"/test_results/%g.xml
+export CMOCKA_XML_FILE="${DAOS_BASE}/test_results/%g"
+[[ -z ${WITH_VALGRIND} ]] && export "${CMOCKA_XML_FILE}.xml" \
```

Comment:
(lint) Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]